### PR TITLE
Clarify Pod/Service DNS a little

### PR DESCRIPTION
**Pull request by thockin**
I found this section to be a litt ambiguous and hard to follow.Z

### CODE REVIEW

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 1]
```diff
@@ -70,29 +71,28 @@ For more up-to-date specification, see
...
+depending on the IP family or families of the Service, with a name of the form
 `my-svc.my-namespace.svc.cluster-domain.example`.  This resolves to the cluster IP
 of the Service.
 
-"Headless" (without a cluster IP) Services are also assigned a DNS A or AAAA record,
-depending on the IP family of the Service, for a name of the form
-`my-svc.my-namespace.svc.cluster-domain.example`.  Unlike normal
-Services, this resolves to the set of IPs of the Pods selected by the Service.
+[Headless Services](/docs/concepts/services-networking/service/#headless-services) 
+(without a cluster IP) Services are also assigned DNS A and/or AAAA records,
```
Comment (reylejano) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:80]: remove duplicate of "Services"
```suggestion
(without a cluster IP) are also assigned DNS A and/or AAAA records,
```

**Comment by reylejano** [comment]
Preview: 
https://deploy-preview-37898--kubernetes-io-main-staging.netlify.app/docs/concepts/services-networking/dns-pod-service/

small [suggestion](https://github.com/kubernetes/website/pull/37898#discussion_r1025724164), otherwise lgtm

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 2]
```diff
@@ -16,13 +17,13 @@ Services with consistent DNS names instead of IP addresses.
...
-the kubelets to tell individual containers to use the DNS Service's IP to
-resolve DNS names.
+Kubernetes publishes information about Pods and Services which is used
+to program DNS.  Kubelet configures Pods' DNS so that running containers
+can lookup Services by name rather than IP.
 
-Every Service defined in the cluster (including the DNS server itself) is
-assigned a DNS name. By default, a client Pod's DNS search list includes the 
-Pod's own namespace and the cluster's default domain. 
+Services defined in the cluster are assigned a DNS name. By default, a
```
Comment (tengqm) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:24]: possible nit (plural issue):

Each Service defined in the cluster is assigned a DNS name.

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 3]
```diff
@@ -113,33 +113,40 @@ Any Pods exposed by a Service have the following DNS resolution available:
...
 metadata:
-  name: default-subdomain
+  name: busybox-subdomain
 spec:
   selector:
     name: busybox
   clusterIP: None
   ports:
-  - name: foo # Actually, no port is needed.
+  - name: foo # name is not required for a single ports
```
Comment (tengqm) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:148]: nit:

name is not required for a single-port Service

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 4]
```diff
@@ -174,32 +181,28 @@ spec:
...
+pointing to the Pods' IP addresses. Also, the Pod needs to be ready in order to have a
 record unless `publishNotReadyAddresses=True` is set on the Service.
 {{< /note >}}
 
 ### Pod's setHostnameAsFQDN field {#pod-sethostnameasfqdn-field}
 
 {{< feature-state for_k8s_version="v1.22" state="stable" >}}
 
-When a Pod is configured to have fully qualified domain name (FQDN), its hostname is the short hostname. For example, if you have a Pod with the fully qualified domain name `busybox-1.default-subdomain.my-namespace.svc.cluster-domain.example`, then by default the `hostname` command inside that Pod returns `busybox-1` and  the `hostname --fqdn` command returns the FQDN.
+When a Pod is configured to have fully qualified domain name (FQDN), its hostname is the short hostname. For example, if you have a Pod with the fully qualified domain name `busybox-1.busybox-subdomain.my-namespace.svc.cluster-domain.example`, then by default the `hostname` command inside that Pod returns `busybox-1` and  the `hostname --fqdn` command returns the FQDN.
```
Comment (tengqm) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:205]: Wrap this long line?

File: content/en/docs/concepts/services-networking/dns-pod-service.md (hunk 2 above)
Comment (thockin) [reply content/en/docs/concepts/services-networking/dns-pod-service.md:24]: You are right!

File: content/en/docs/concepts/services-networking/dns-pod-service.md (hunk 4 above)
Comment (thockin) [reply content/en/docs/concepts/services-networking/dns-pod-service.md]: ACK - whole doc could use a linewrap or ten

**Comment by thockin** [comment]
Re-pushed

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 5]
```diff
@@ -16,13 +17,13 @@ Services with consistent DNS names instead of IP addresses.
 
 <!-- body -->
 
-Kubernetes DNS schedules a DNS Pod and Service on the cluster, and configures
-the kubelets to tell individual containers to use the DNS Service's IP to
-resolve DNS names.
+Kubernetes publishes information about Pods and Services which is used
+to program DNS.  Kubelet configures Pods' DNS so that running containers
```
Comment (kbhawkey) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:21]: Hi @thockin .
Does Kubernetes 'program DNS', or provide or implement DNS?

nit: At the end of a sentence after a period, use a single space.

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 6]
```diff
@@ -16,13 +17,13 @@ Services with consistent DNS names instead of IP addresses.
...
+can lookup Services by name rather than IP.
 
-Every Service defined in the cluster (including the DNS server itself) is
-assigned a DNS name. By default, a client Pod's DNS search list includes the 
-Pod's own namespace and the cluster's default domain. 
+Services defined in the cluster are assigned DNS names. By default, a
```
Comment (kbhawkey) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:24]: Who or What assigns the DNS names?

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 7]
```diff
@@ -39,7 +40,7 @@ A query for `data.prod` returns the intended result, because it specifies the
 namespace. 
 
 DNS queries may be expanded using the Pod's `/etc/resolv.conf`. Kubelet 
-sets this file for each Pod. For example, a query for just `data` may be 
+configures this file for each Pod. For example, a query for just `data` may be
```
Comment (kbhawkey) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:43]: nit: Instead of `this` use `/etc/resolv.conf`.

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 8]
```diff
@@ -70,29 +71,28 @@ For more up-to-date specification, see
...
 Clients are expected to consume the set or else use standard round-robin
 selection from the set.
 
 ### SRV records
 
-SRV Records are created for named ports that are part of normal or [Headless
-Services](/docs/concepts/services-networking/service/#headless-services).
-For each named port, the SRV record would have the form
-`_my-port-name._my-port-protocol.my-svc.my-namespace.svc.cluster-domain.example`.
+SRV Records are created for named ports that are part of normal or headless
```
Comment (kbhawkey) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:88]: Kubernetes creates Service records (SRV records) for named ports that are part of normal or headless services.

File: content/en/docs/concepts/services-networking/dns-pod-service.md (hunk 6 above)
Comment (sftim) [reply content/en/docs/concepts/services-networking/dns-pod-service.md:24]: It's definitely helpful to be clear

File: content/en/docs/concepts/services-networking/dns-pod-service.md (hunk 5 above)
Comment (sftim) [reply content/en/docs/concepts/services-networking/dns-pod-service.md:21]: DNS **is** a separate concern (it's usually, but not always, CoreDNS that does the work).
Maybe “configure cluster DNS”, where cluster DNS could be italicized or even be a glossary tooltip.

**Review by sftim — COMMENTED** [review]
/hold

/lgtm
to me, as someone who has a CKA at least
and to me, as a SIG Docs reviewer.

There's room for more tweaks - hence the hold - but at the same time these changes leave the page better than when they found it, and I don't see any dealbreaker regressions.

@thockin what we like to do when there are outstanding docs nits is one of:
- address the nits
- you are welcome to unhold, and then please file at least one Good First Issue against k/website that describes a remaining nit that you agree is good to fix.

**Comment by sftim** [comment]
@kbhawkey what do you recommend we do here? The outstanding feedback is points you raised.

**Comment by kbhawkey** [comment]
> @kbhawkey what do you recommend we do here? The outstanding feedback is points you raised.

My feedback is mostly related to clarity. I found that if some of the content is not as clear as it could be. We can revisit the content again. OK to merge!
